File: packages/cli/src/commands/spec.js
import { CollectionManager, FileStorage } from '@apiforge/core';

/**
 * apiforge spec <subcommand>
 *   list             - List imported specs/collections
 *   show <name>      - Show endpoints in a collection
 */
export async function specCommand(args, dataDir) {
  const sub = args[0];
  const storage = new FileStorage(`${dataDir}/data`);
  const collectionManager = new CollectionManager(storage);

  switch (sub) {
    case 'list': {
      const collections = await collectionManager.list();
      if (collections.length === 0) {
        console.log('No specs imported. Import one with: apiforge import <file|url>');
        return;
      }
      console.log('Imported specs:');
      for (const col of collections) {
        console.log(`  ${col.name}`);
      }
      break;
    }

    case 'show': {
      const name = args[1];
      if (!name) { console.error('Usage: apiforge spec show <name>'); process.exit(1); }
      const col = await collectionManager.getByName(name);
      if (!col) { console.error(`Not found: ${name}`); process.exit(1); }

      console.log(`${col.name} (v${col.version})`);
      if (col.description) console.log(`  ${col.description}`);
      console.log('');

      for (const item of col.items) {
        if (item.type === 'folder') {
          console.log(`  [${item.name}]`);
          for (const req of item.items || []) {
            if (req.type === 'request' && req.request) {
              const method = req.request.method.padEnd(7);
              console.log(`    ${method} ${req.request.name}`);
            }
          }
        } else if (item.type === 'request' && item.request) {
          const method = item.request.method.padEnd(7);
          console.log(`  ${method} ${item.request.name}`);
        }
      }
      break;
    }

    default:
      console.error('Usage: apiforge spec <list|show>');
      process.exit(1);
  }
}

File: packages/apiforge-rs/crates/apiforge-cli/src/commands/spec.rs
use anyhow::Result;
use apiforge_core::collection::CollectionManager;
use crossterm::style::Stylize;

use crate::SpecAction;

pub fn execute(action: SpecAction) -> Result<()> {
    let manager = CollectionManager::new()?;

    match action {
        SpecAction::List => {
            let collections = manager.list()?;
            if collections.is_empty() {
                println!("{}", "No specs imported. Import one with: apiforge import <file|url>".dark_grey());
            } else {
                println!("{}", "Imported specs:".bold());
                for col in &collections {
                    let version = col.version.as_deref().unwrap_or("");
                    let desc = col.description.as_deref().unwrap_or("");
                    print!("  {} {}", "●".green(), col.name.clone().bold());
                    if !version.is_empty() {
                        print!(" (v{})", version);
                    }
                    println!();
                    if !desc.is_empty() {
                        println!("    {}", desc.dark_grey());
                    }
                }
            }
        }
        SpecAction::Show { name } => {
            let collection = manager.get_by_name(&name)?;
            match collection {
                Some(col) => {
                    let version = col.version.as_deref().unwrap_or("?");
                    println!("{}", format!("{} (v{})", col.name, version).bold());
                    if let Some(desc) = &col.description {
                        println!("  {}", desc);
                    }
                    println!();

                    for item in &col.items {
                        if item.item_type == "folder" {
                            println!("  {}", format!("[{}]", item.name).cyan());
                            if let Some(sub_items) = &item.items {
                                for req in sub_items {
                                    if req.item_type == "request" {
                                        if let Some(r) = &req.request {
                                            println!("    {} {}", format!("{:<7}", r.method).bold(), r.name);
                                        }
                                    }
                                }
                            }
                        } else if item.item_type == "request" {
                            if let Some(r) = &item.request {
                                println!("  {} {}", format!("{:<7}", r.method).bold(), r.name);
                            }
                        }
                    }
                }
                None => {
                    eprintln!("{}", format!("Not found: {}", name).red());
                }
            }
        }
    }
    Ok(())
}
